internal/service/imagebuilder/container_recipe_test.go

```diff
@@ -540,6 +540,183 @@ func TestAccImageBuilderContainerRecipe_kmsKeyID(t *testing.T) {
 	})
 }
 
+func TestAccImageBuilderContainerRecipe_platform_override_linux(t *testing.T) {
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	resourceName := "aws_imagebuilder_container_recipe.test"
+	ecrImage := acctest.ImageBuilderECRImageFromEnv(t, imagebuilder.PlatformLinux)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acctest.PreCheck(t) },
+		ErrorCheck:        acctest.ErrorCheck(t, imagebuilder.EndpointsID),
+		ProviderFactories: acctest.ProviderFactories,
+		CheckDestroy:      testAccCheckContainerRecipeDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccContainerRecipePlatformOverrideLinuxConfig(rName, ecrImage),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckContainerRecipeExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "parent_image", ecrImage),
+					resource.TestCheckResourceAttr(resourceName, "platform_override", imagebuilder.PlatformLinux),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"platform_override"},
+			},
+		},
+	})
+}
+
+func TestAccImageBuilderContainerRecipe_platform_override_linux_wrong_parent_image(t *testing.T) {
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	resourceName := "aws_imagebuilder_container_recipe.test"
+	parentImage := "arn:${data.aws_partition.current.partition}:imagebuilder:${data.aws_region.current.name}:aws:image/amazon-linux-x86-2/x.x.x"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acctest.PreCheck(t) },
+		ErrorCheck:        testAccErrorCheckSkipContainerRecipe(t),
+		ProviderFactories: acctest.ProviderFactories,
+		CheckDestroy:      testAccCheckContainerRecipeDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccContainerRecipePlatformOverrideLinuxConfig(rName, parentImage),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckContainerRecipeExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "parent_image", parentImage),
+					resource.TestCheckResourceAttr(resourceName, "platform_override", imagebuilder.PlatformLinux),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"platform_override"},
+			},
+		},
+	})
+}
+
+func TestAccImageBuilderContainerRecipe_platform_override_linux_without_override(t *testing.T) {
```

Comment:
This test is FAILing for me - I think it needs the same `ExpectError` treatment as above.